### ISSUE TITLE
Some suggestions and Improvements

### ISSUE THREAD

**Issue by aneesh1122** [ErrorFlynn/ytdlp-interface#132]
1) Replace 'convert audio to mp3' to 'only audio'. Those who need mp3 can set their preferred audio container as mp3 in settings

2) Add an option to crop thumbnail to square(1:1). Music have square album covers and thumbnail in youtube is 16:9 always. when downloading music from youtube it extracts the default thumbnail which is 16:9. Cropping it will give the actual album cover
Here is the code you can use for cropping
`--ppa "ThumbnailsConvertor+ffmpeg_o:-c:v png -vf crop='ih'"`

3) add the option to add metadata

4) add 'mkv' in the list of preferred video container

5) Remove this in the 'select formats' page
![Screenshot (364)](https://github.com/ErrorFlynn/ytdlp-interface/assets/45353488/6b6d2ec9-8832-4481-9f8d-e8d10447843d)

when we select multiple audio streams it should automatically add '--audio-multistreams' in the code

6) Most of the times videos have auto generated lyrics. when 'Embed subtitles' is selected it should embed auto generated subtitles if the uploader's original subtitle is not available. Here is the code to do that
`--write-subs --write-auto-subs --embed-subs --compat-options no-keep-subs `
this code will prioritize uploader's sub over auto sub

7) Add option to add playlist range to download. When this option is selected add `-I` in the code and take input from user for range. something like this
![Screenshot (365)](https://github.com/ErrorFlynn/ytdlp-interface/assets/45353488/7a707be8-1c10-4ac7-8bac-478e4940aa70)
Add this text as the description when cursor in on it
"1:3,7,-5::2" used on a playlist of size 15 will download the items at index 1,2,3,7,11,13,15"

**Comment by ErrorFlynn** [comment]
>Add option to add playlist range to download.

That feature already exists, it sounds like you might not be aware of it. Select a queue item that's a playlist, and then 
`[queue menu]`->`Select videos`.

**Comment by aneesh1122** [comment]
> >Add option to add playlist range to download.
> 
> That feature already exists, it sounds like you might not be aware of it. Select a queue item that's a playlist, and then 
> `[queue menu]`->`Select videos`.

I wasn't aware of that. Sorry 😅

That's a great way of handling playlists. Will try it now.

What do you think about the rest of the suggestions?

**Comment by ErrorFlynn** [comment]
>What do you think about the rest of the suggestions?

I haven't had time to properly consider most of them (I'll get back to you on that), but I can tell you that an mkv preferred container option sounds reasonable (it would have to be implemented by passing `--merge-output-format mkv` to yt-dlp).

**Comment by aneesh1122** [comment]
> >What do you think about the rest of the suggestions?
> 
> I haven't had time to properly consider most of them (I'll get back to you on that), but I can tell you that an mkv preferred container option sounds reasonable (it would have to be implemented by passing `--merge-output-format mkv` to yt-dlp).

Yeah. mkv is compatible with almost everything and also supported by almost every player.

Btw I found out why I never got to know about selecting videos from a playlist. I was downloading from a YouTube music playlist. It looks like selecting videos is not supported with a YouTube music playlist.
It's not a problem though since I can always use the same playlist from YouTube. Just wanted to tell you about incompatibility with YouTube music.
